packages/plugin-credentials/test/automation/spec/authorization-code-grant.js

```diff
@@ -0,0 +1,71 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {createBrowser} from '@ciscospark/test-helper-automation';
+import testUsers from '@ciscospark/test-helper-test-users';
+import pkg from '../../../package.json';
+
+const redirectUri = process.env.CISCOSPARK_REDIRECT_URI || process.env.REDIRECT_URI;
+
+describe(`spark-core`, function() {
+  this.timeout(120000);
+  describe(`Authorization`, () => {
+    describe(`Authorization Code Grant`, () => {
+      let browser, user;
+
+      before(() => testUsers.create({count: 1})
+        .then((users) => {
+          user = users[0];
+        }));
+
+      before(() => createBrowser(pkg)
+        .then((b) => {
+          browser = b;
+        }));
+
+      after(() => browser.printLogs());
+
+      after(() => browser && browser.quit()
+        .catch((reason) => {
+          console.warn(reason);
+        }));
+
+      it(`authorizes a user`, () => browser
+        .get(redirectUri)
+        .waitForElementByClassName(`ready`)
+        .title()
+          .should.eventually.become(`Authorization Automation Test`)
+        .waitForElementByCssSelector(`[title="Login with Authorization Code Grant"]`)
+          .click()
+        .login(user)
+        .waitForElementByClassName(`authorization-automation-test`)
+        .waitForElementById(`refresh-token`)
+          .text()
+            .should.eventually.not.be.empty
```

Comment:
Yea, I'm actually thinking we'll need to `credentials-spec` package in the near future. this plugin does a lot of stuff that the credentials plugin in http-core doesn't need to, but both really need to have the same interface.